test/tests/functional/pbs_execjob_susp_resume.py

```diff
@@ -0,0 +1,546 @@
+                          'use -p moms=<mom1>:<mom2>')
+        TestFunctional.setUp(self)
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+
+        # execjob_postsuspend hook
+        self.postsuspend_hook_body = """import pbs
+e=pbs.event()
+
+def proc_status(pid):
+    for line in open("/proc/%d/status" % pid).readlines():
+        if line.startswith("State:"):
+            return line.split(":",1)[1].strip().split(' ')[0]
+    return None
+
+def print_attribs(pbs_obj, header):
+    for a in pbs_obj.attributes:
+        v = getattr(pbs_obj, a)
+        if v and str(v) != "":
+            pbs.logjobmsg(e.job.id, "%s: %s = %s" % (header, a, v))
+            if a == "session_id":
+                st = proc_status(v)
+                if st == 'T':
+                    pbs.logjobmsg(e.job.id,
+                                  "%s: process seen as suspended" % header)
+
+if e.type == pbs.EXECJOB_POSTSUSPEND:
+    pbs.logmsg(pbs.LOG_DEBUG, "called execjob_postsuspend hook")
+print_attribs(e.job, "JOB")
+
+for vn in e.vnode_list:
+    v = e.vnode_list[vn]
+    print_attribs(v, "vnode_list[" + vn + "]")
+"""
+        # execjob_postsuspend hook, reject
+        self.postsuspend_hook_reject_body = """import pbs
+e=pbs.event()
+job=e.job
+e.reject("bad suspend")
+"""
+        # execjob_postsuspend hook, reject by sister only
+        self.postsuspend_hook_sis_reject_body = """import pbs
+e=pbs.event()
+job=e.job
+if not e.job.in_ms_mom():
+    e.reject("bad suspend")
+"""
+        # hook with an unhandled exception
+        self.hook_error_body = """import pbs
+e=pbs.event()
+job=e.job
+raise NameError
+"""
+        # hook with an unhandled exception, sister only
+        self.hook_sis_error_body = """import pbs
+e=pbs.event()
+job=e.job
+if not job.in_ms_mom():
+    raise NameError
+"""
+
+        # execjob_preresume hook
+        self.preresume_hook_body = """import pbs
+e=pbs.event()
+
+def proc_status(pid):
+    for line in open("/proc/%d/status" % pid).readlines():
+        if line.startswith("State:"):
+            return line.split(":",1)[1].strip().split(' ')[0]
+    return None
+
+def print_attribs(pbs_obj, header):
+    for a in pbs_obj.attributes:
+        v = getattr(pbs_obj, a)
+        if v and str(v) != "":
+            pbs.logjobmsg(e.job.id, "%s: %s = %s" % (header, a, v))
+            if a == "session_id":
+                st = proc_status(v)
+                if st == 'T':
+                    pbs.logjobmsg(e.job.id,
+                                  "%s: process seen as suspended" % header)
+
+if e.type == pbs.EXECJOB_PRERESUME:
+    pbs.logmsg(pbs.LOG_DEBUG, "called execjob_preresume hook")
+
+print_attribs(e.job, "JOB")
+
+for vn in e.vnode_list:
+    v = e.vnode_list[vn]
+    print_attribs(v, "vnode_list[" + vn + "]")
+"""
+        # execjob_preresume hook, reject
+        self.preresume_hook_reject_body = """import pbs
+e=pbs.event()
+job=e.job
+e.reject("bad resumption")
```

Comment:
You may want to change these to "bad resumption on ms" and "bad resumption on sis".